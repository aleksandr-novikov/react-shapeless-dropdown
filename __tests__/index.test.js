import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Dropdown from '../src';

Enzyme.configure({adapter: new Adapter()});
const dummy = ({text}) => (<button className={text}>{text}</button>);

describe('Dropdown component', () => {
    test('Renders without props', () => {
        const wrapper = shallow(<Dropdown/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders with text button and text content', () => {
        const wrapper = shallow(<Dropdown button={'Menu'}>Content</Dropdown>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders with component button and component content', () => {
        const wrapper = shallow(<Dropdown
            button={dummy({text: 'Menu'})}>{dummy({text: 'Content'})}</Dropdown>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Handles clicks on button and changes state. Fires callbacks', () => {
        const openCb = jest.fn();
        const closeCb = jest.fn();
        const wrapper = mount(<Dropdown onDropdownOpen={openCb}
                                        onDropdownClose={closeCb}
                                        button={dummy({text: 'Menu'})}/>);
        const button = wrapper.find('button');

        button.simulate('click');
        expect(wrapper.state('opened')).toBe(true);
        button.simulate('click');
        expect(wrapper.state('opened')).toBe(false);

        expect(openCb).toHaveBeenCalled();
        expect(closeCb).toHaveBeenCalled();

    });

    test('Closes dropdown on scroll and on action', () => {
        const wrapper = mount(<Dropdown closeOnScroll={true}
                                        closeOnAction={true}
                                        button={dummy({text: 'Menu'})}/>);

        wrapper.setState({opened: true});
        global.dispatchEvent(new Event('scroll'));
        expect(wrapper.state('opened')).toBe(false);

        wrapper.setState({opened: true});
        wrapper.find('div').findWhere(n => typeof n.prop('onKeyDown') === 'function')
            .at(1).simulate('keyDown', {key: 'Enter'});
        expect(wrapper.state('opened')).toBe(false);
    });

    test('Updates its bounds on resize', () => {
        const wrapper = mount(<Dropdown button={'Menu'}/>);
        const width = 500;

        wrapper.setState({
            dropdownBounds: {width}
        });
        global.dispatchEvent(new Event('resize'));
        expect(wrapper.state('dropdownBounds').width !== width);

    });

});
