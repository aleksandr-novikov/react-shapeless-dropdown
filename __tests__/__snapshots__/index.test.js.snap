// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component Renders with component button and component content 1`] = `ShallowWrapper {}`;

exports[`Dropdown component Renders with text button and text content 1`] = `ShallowWrapper {}`;

exports[`Dropdown component Renders without props 1`] = `ShallowWrapper {}`;
